tardis/energy_input/gamma_ray_grid.py

```diff
@@ -0,0 +1,226 @@
+import numpy as np
+from astropy.coordinates import cartesian_to_spherical
+import re
+from nuclear.io.nndc import get_decay_radiation_database, store_decay_radiation
+import pandas as pd
+
+from tardis.energy_input.util import (
+    solve_quadratic_equation,
+    convert_half_life_to_astropy_units,
+)
+from tardis.util.base import atomic_number2element_symbol
+import tardis.constants as const
+
+
+def calculate_distance_radial(gxpacket, r_inner, r_outer):
```

Comment:
use https://github.com/tardis-sn/tardis/blob/master/tardis/montecarlo/montecarlo_numba/calculate_distances.py#L21 as a template to restructure this and make more clear for which cases